src/User32.Desktop/User32.cs

```diff
@@ -621,10 +674,478 @@ public static partial class User32
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SwitchDesktop(SafeDesktopHandle hDesktop);
+
+        /// <summary>
+        /// Assigns the specified desktop to the calling thread. All subsequent operations on the desktop use the access rights granted to the desktop.
+        /// </summary>
+        /// <param name="hDesktop">
+        /// A handle to the desktop to be assigned to the calling thread.
+        /// This handle is returned by the <see cref="CreateDesktop(string, string, IntPtr, DesktopCreationFlags, DESKTOP_ACCESS_MASK, Kernel32.SECURITY_ATTRIBUTES*)"/>, <see cref="GetThreadDesktop"/>, <see cref="OpenDesktop"/>, or <see cref="OpenInputDesktop"/> function.
+        /// This desktop must be associated with the current window station for the process.
+        /// </param>
+        /// <returns>If the function succeeds, the return value is true, if it fails, the return value is false.</returns>
+        /// <remarks>
+        /// The function will fail if the calling thread has any windows or hooks on its current desktop (unless the <paramref name="hDesktop"/> parameter is a handle to the current desktop).
+        /// </remarks>
+        [DllImport(nameof(User32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SetThreadDesktop(SafeDesktopHandle hDesktop);
+
+        /// <summary>
+        /// Opens the specified desktop object.
+        /// </summary>
+        /// <param name="lpszDesktop">The name of the desktop to be opened. Desktop names are case-insensitive. This desktop must belong to the current window station.</param>
+        /// <param name="dwFlags">Access control flags</param>
+        /// <param name="fInherit">If this value is true, processes created by this process will inherit the handle. Otherwise, the processes do not inherit this handle.</param>
+        /// <param name="dwDesiredAccess">The access to the desktop. For a list of access rights, see <see cref="DESKTOP_ACCESS_MASK"/>.</param>
+        /// <returns>If the function succeeds, the return value is a handle to the opened desktop, if the function fails, the return value is an invalid handle.</returns>
+        /// <remarks>
+        /// <para>
+        /// When you are finished using the handle, call the <see cref="CloseDesktop"/> function to close it.
+        /// </para>
+        /// <para>
+        /// The calling process must have an associated window station, either assigned by the system at process creation time or set by the <see cref="SetProcessWindowStation"/> function.
+        /// </para>
+        /// <para>
+        /// If the <paramref name="dwDesiredAccess"/> parameter specifies the READ_CONTROL, WRITE_DAC, or WRITE_OWNER standard access rights, you must also request the DESKTOP_READOBJECTS and DESKTOP_WRITEOBJECTS access rights.
+        /// </para>
+        /// </remarks>
+        [DllImport(nameof(User32), SetLastError = true)]
```

Comment:
We try to specify `CharSet = Unicode` on all p/invoke methods that use various character types in their signature like this one does. I'm not sure if that's necessary, but I think it's slightly faster if .NET doesn't have to look for various forms of the exported method.